File: backend/Mover.Entities/POCOEntities/Georeferencia/Departamento.cs

namespace Mover.Entities.POCOEntities.Georeferencia
{
    public class Departamento
    {
        public int Id { get; set; }
        public string Codigo { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public int PaisId { get; set; } 
        public Pais? Pais { get; set; }
    }
}

File: frontend/lib/domain/entities/georeferencia/departamento.dart
class Departamento {
  final int? id;
  final int? paisId;
  final String nombre;
  final String codigo;

  Departamento({
    this.id,
    this.paisId,
    this.nombre = '',
    this.codigo = '',
  });

  factory Departamento.fromJson(Map<String, dynamic> json) {
    return Departamento(
        id: json['id'],
        nombre: json['nombre'],
        paisId: json['paisId'],
        codigo: json['codigo']);
  }

  Map<String, dynamic> toJson() {
    return {'id': id, 'nombre': nombre, 'paisId': paisId, 'codigo': codigo};
  }
}
